laika/reports.py

```diff
@@ -1248,6 +1305,34 @@ def save(self):
         self.s3.put_object(Bucket=self.bucket, Key=filename, Body=data)
 
 
+class FixedColumnarResult(Result):
+    """
+    Wrapper result that ensures the presence of a list of columns in the data
+    before sending them to an inner result. If a column is not present in the
+    data, a column is added and filled with some value (np.nan by default).
+
+    The data is expected to be a pandas.DataFrame or should be acceptable
+    by the DataFrame's constructor.
+    """
+
+    columns = []
+    default_value = np.nan
+
+    def __init__(self, conf, data, **kwargs):
+        super(FixedColumnarResult, self).__init__(conf, data, **kwargs)
+
+        if not isinstance(data, pd.DataFrame):
+            data = pd.DataFrame(self.data)
+        for column in self.columns:
+            data[column] = data.get(column, self.default_value)
+
+        inner_result_class = conf.get_result_class(self.inner_result_type)
+        self._inner_result = inner_result_class(conf, data[self.columns], **kwargs)
```

Comment:
I think that you should mention on the documentation that *only* the columns defined on the documentation will be kept or leave all other columns that are present on the former data passed to this result.

Reply:
👍 Clarified a bit in https://github.com/trocafone/laika/pull/22/commits/8c287d403dcbfc3d63f5b6870b6c57ae5bc1ca76